File: geometry/distance_formula/python/distance_formula.py
import math

def calculate_distance(u, v):
    """Distance between points u and v, where u and v are points in n dimensional
    space, denoted by lists/tuples/arrays.
    Usage: calculate_distance( <first point>, <second point>)
    """

    if len(u) != len(v):
        raise ValueError('Dimensionality of points doesn\'t match')

    retVal = 0
    for i in xrange(len(u)):
        retVal += (u[i] - v[i])**2

    return math.sqrt(retVal)

File: geometry/distance_formula/cpp/distance_formula.cpp
#include <iostream>
#include <math.h>

using namespace std;

double calculate_distance(pair<int, int> u, pair<int, int> v) {
  return sqrt((u.first - v.first)*(u.first - v.first) + (u.second - v.second)*(u.second - v.second));
}

int main() {
  pair<int, int> u = make_pair(5, 4);
  pair<int, int> v = make_pair(5, 5);
  cout << calculate_distance(u, v);
  return 0;
}
